src/goto-cc/linker_script_merge.cpp

```diff
@@ -453,24 +446,46 @@ int linker_script_merget::ls_data2instructions(
 
     constant_exprt    array_size_expr=from_integer(array_size, size_type());
     array_typet       array_type(char_type(), array_size_expr);
-    symbol_exprt      array_expr(array_name.str(), array_type);
-    source_locationt  array_loc;
 
+    source_locationt array_loc;
     array_loc.set_file(linker_script);
     std::ostringstream array_comment;
     array_comment << "Object section '" << d["section"].value << "' of size "
                   << array_size << " bytes";
     array_loc.set_comment(array_comment.str());
-    array_expr.add_source_location()=array_loc;
+
+    namespacet ns(symbol_table);
+    const auto zi = zero_initializer(array_type, array_loc, ns);
+    CHECK_RETURN(zi.has_value());
+
+    // Add the array to the symbol table.
+    symbolt array_sym;
```

Comment:
Mildly a shame there isn't a constructor/utility function for creating a suitably initialised `symbolt`

Reply:
Thank you for calling this out! I'll try to address this in a separate PR.